src/nccl_ofi_rdma.c

```diff
@@ -1676,98 +1670,114 @@ static int process_pending_reqs(nccl_net_ofi_rdma_ep_t *ep)
 	return rc;
 }
 
-/*
- * @brief	Process completion entries for the given completion quque.
- *		This also updates several request fileds like size, status, etc
- *
- * @return	0, on success
- *		error, on others
- */
-static int ofi_process_cq(nccl_net_ofi_rdma_ep_t *ep)
+static int ofi_process_cq_rail(nccl_net_ofi_rdma_ep_t *ep, nccl_net_ofi_ep_rail_t *rail)
 {
 	ssize_t rc = 0;
 	int ret = 0;
 	struct fi_cq_err_entry err_buffer = { 0 };
 	struct fi_cq_tagged_entry cqe_tagged_buffers[cq_read_count];
 	nccl_net_ofi_rdma_req_t *req = NULL;
 
-	for (int rail_id = 0; rail_id != ep->num_rails; ++rail_id) {
-		nccl_net_ofi_ep_rail_t *rail = get_rail(ep, rail_id);
-
-		while (true) {
-			/* Receive completions for the given endpoint */
-			rc = fi_cq_read(rail->cq, cqe_tagged_buffers, cq_read_count);
-			if (rc > 0) {
-				ret = process_completions(
-					cqe_tagged_buffers, rc, ep, rail_id);
-				if (OFI_UNLIKELY(ret != 0))
-					goto exit;
-			} else if (OFI_UNLIKELY(rc == -FI_EAVAIL)) {
-				rc = fi_cq_readerr(rail->cq, &err_buffer, 0);
-				if (OFI_UNLIKELY(rc == -FI_EAGAIN)) {
-					/*
-					 * Error not available yet.
-					 * fi_cq_read will keep returning -FI_EAVAIL so just bail out and try again later.
-					 */
-					break;
-				} else if (OFI_UNLIKELY(rc < 0)) {
-					NCCL_OFI_WARN("Unable to read from fi_cq_readerr. RC: %zd. Error: %s",
-						      rc,
-						      fi_strerror(-rc));
+	while (true) {
+		/* Receive completions for the given endpoint */
+		rc = fi_cq_read(rail->cq, cqe_tagged_buffers, cq_read_count);
+		if (rc > 0) {
+			ret = process_completions(
+				cqe_tagged_buffers, rc, ep, rail);
+			if (OFI_UNLIKELY(ret != 0))
+				goto exit;
+		} else if (OFI_UNLIKELY(rc == -FI_EAVAIL)) {
+			rc = fi_cq_readerr(rail->cq, &err_buffer, 0);
+			if (OFI_UNLIKELY(rc == -FI_EAGAIN)) {
+				/*
+				 * Error not available yet.
+				 * fi_cq_read will keep returning -FI_EAVAIL so just bail out and try again later.
+				 */
+				break;
```

Comment:
I don't think this will happen (cq_read failing without an error entry enqueued in the error CQ), unless there was no error CQE to be written in the first place which wouldn't have gotten into this path.

Reply:
That code is in both the sendrecv and rdma completion code.  I think you're right, but this is only an indentation change, so let's deal with that later.